### PR TITLE
Port serial tests to catch

### PR DESCRIPTION
**Pull request by connojd**

### CODE REVIEW

File: src/serial/test/test_serial_port_intel_x64.cpp [hunk 1]
```diff
@@ -22,193 +22,241 @@
...
+
+    CHECK(serial_port_intel_x64::instance()->port() == DEFAULT_COM_PORT);
+    CHECK(serial_port_intel_x64::instance()->baud_rate() == serial_port_intel_x64::DEFAULT_BAUD_RATE);
+    CHECK(serial_port_intel_x64::instance()->data_bits() == serial_port_intel_x64::DEFAULT_DATA_BITS);
+    CHECK(serial_port_intel_x64::instance()->stop_bits() == serial_port_intel_x64::DEFAULT_STOP_BITS);
+    CHECK(serial_port_intel_x64::instance()->parity_bits() ==
+          serial_port_intel_x64::DEFAULT_PARITY_BITS);
+
+    CHECK((g_ports[DEFAULT_COM_PORT + serial_intel_x64::baud_rate_lo_reg]) == ((
+                serial_port_intel_x64::DEFAULT_BAUD_RATE & 0x00FF) >> 0));
```
Comment (rianquinn) [inline src/serial/test/test_serial_port_intel_x64.cpp]: whitespace is off

File: src/serial/test/CMakeLists.txt [hunk 2]
```diff
@@ -6,12 +6,24 @@ include("${CMAKE_INSTALL_PREFIX}/cmake/CMakeGlobal_Includes.txt")
...
-    add_executable(test_${str} test_${str}.cpp )
+macro(do_test str source)
+    add_executable(test_${str} test_${str}.cpp ${source})
+    target_link_libraries(test_${str} bfvmm_intrinsics_static)
     add_test(test_${str} test_${str})
 endmacro(do_test)
 
-do_test(serial_port_intel_x64)
+list(APPEND SERIAL_SOURCE
+    ../src/serial_port_intel_x64.cpp
```
Comment (rianquinn) [inline src/serial/test/CMakeLists.txt]: you did this perfectly. serial_port_intel_x64.cpp is added manually while the supporting libraries like intrinsics that are not being tested are statically linked. Great job!!!

File: src/serial/test/test_serial_port_intel_x64.cpp [hunk 3]
```diff
@@ -22,193 +22,241 @@
...
+TEST_CASE("serial: set_parity_bits_success_extra_bits")
 {
-    CHECK(true);
+    MockRepository mocks;
+    mock_serial(mocks);
+
+    auto serial = std::make_unique<serial_port_intel_x64>();
+
+    auto bits = serial_port_intel_x64::DEFAULT_PARITY_BITS |
+                ~serial_intel_x64::line_control_parity_mask;
```
Comment (rianquinn) [inline src/serial/test/test_serial_port_intel_x64.cpp]: if astyle will let you, I prefer 4 spaces instead of aligned with =.

**Comment by rianquinn** [comment]
@connojd the code coverage tests are failing for LLVM because inline headers still get sucked in. There are however some issues specific to serial that LLVM found. Once that above comments are fixed, and LLVM is only complaining about headers in other modules, I'll merge.

File: src/serial/test/test_serial_port_intel_x64.cpp (hunk 3 above)
Comment (connojd) [inline src/serial/test/test_serial_port_intel_x64.cpp]: both of these are from astyle

**Comment by connojd** [comment]
@rianquinn spacing fixed

**Comment by rianquinn** [comment]
@connojd spacing looks good but you used target_link_libraries(test_${str} bfvmm_serial_static) to add the serial code instead of adding the cpp files manually. The result is LLVM's checks will ignore the serial code your unit testing (which means it will always pass). 

Please see the following for an example. As you can see here, I add the debug_ring.cpp manually so that it is being tested. 
https://github.com/Bareflank/bfvmm/blob/master/src/debug_ring/test/CMakeLists.txt

**Comment by connojd** [comment]
@rianquinn fixed

**Pull request merged**